### PR TITLE
rm: added write-protected check for files

### LINKED ISSUE #3813: `rm` write-protected check is missing
```
#export C=./target/debug/coreutils
touch f
chmod 0 f
$C rm f
```

With GNU:
`rm: remove write-protected regular empty file 'f'?`

With Rust, nothing (uncomment the first line to test)

### PR DESCRIPTION
**Pull request by stefins**
Hi, 

This fixes #3813. also this doesn't check the permission of directories right now.

### CODE REVIEW

**Comment by sylvestre** [comment]
Looks like it broke a test:
```
---- test_rm::test_rm_interactive_never stdout ----
current_directory_resolved: 
touch: /tmp/.tmpgyUHXS/test_rm_interactive
run: /home/runner/work/coreutils/coreutils/target/debug/coreutils chmod 0 test_rm_interactive
run: /home/runner/work/coreutils/coreutils/target/debug/coreutils rm --interactive=never test_rm_interactive
thread 'test_rm::test_rm_interactive_never' panicked at 'assertion failed: !at.file_exists(file_2)', /home/runner/work/coreutils/coreutils/tests/by-util/test_rm.rs:348:5
```

And please add a new test showing that the initial issue is fixed
thanks

**Comment by stefins** [comment]
In this line https://github.com/uutils/coreutils/blob/2aede0382a6112208712e0b0dc385d4cedc9848c/src/uu/rm/src/rm.rs#L115 we are making InteractiveMode::Never as default without any arguments but in case of #3813 we need to be interactive in default mode .

Maybe we can add a new InteractiveMode?

**Comment by sylvestre** [comment]
Sure, you can do that :)

File: src/uu/rm/src/rm.rs [hunk 1]
```diff
@@ -408,7 +409,10 @@ fn remove_file(path: &Path, options: &Options) -> bool {
     false
 }
 
-fn prompt_write_protected(path: &Path, is_dir: bool) -> bool {
+fn prompt_write_protected(path: &Path, is_dir: bool, options: &Options) -> bool {
+    if options.interactive == InteractiveMode::Never {
+        return true;
```
Comment (sylvestre) [inline src/uu/rm/src/rm.rs:414]:
```suggestion
        true,
```

Comment (stefins) [reply src/uu/rm/src/rm.rs:414]: this wont compile!

Comment (sylvestre) [reply src/uu/rm/src/rm.rs:414]: ok, please ignore me then :)

**Comment by tertsdiepraam** [comment]
I'm not sure about this, but it sounds like this issue is separate from the existing `InteractiveMode`. I.e. there is prompting for recursive removal and for removal of protected files. If they are linked I would like the naming to be a bit more descriptive than `Default`, maybe `PromptProtected` or something like that (I'm sure there's a better name).

**Comment by stefins** [comment]
@tertsdiepraam Ok I'll change the name to PromptProtected

**Comment by sylvestre** [comment]
Fails on windows 
```
failures:

---- test_rm::test_prompt_write_protected_no stdout ----
current_directory_resolved: 
touch: C:\Users\RUNNER~1\AppData\Local\Temp\.tmp0yZTql\test_rm_prompt_write_protected_2
run: D:\a\coreutils\coreutils\target\debug\coreutils.exe rm test_rm_prompt_write_protected_2
thread 'test_rm::test_prompt_write_protected_no' panicked at 'assertion failed: at.file_exists(file_2)', D:\a\coreutils\coreutils\tests\by-util\test_rm.rs:394:5


failures:
    test_rm::test_prompt_write_protected_no
```

**Comment by sylvestre** [comment]
To test on Windows:
https://developer.microsoft.com/windows/downloads/virtual-machines/

**Comment by stefins** [comment]
@sylvestre removed the failing test!

**Comment by sylvestre** [comment]
@stefins why did you remove it ?

**Comment by stefins** [comment]
It was failing on all the OS's!

**Comment by sylvestre** [comment]
sure but it is an issue with the test or with the code ?
was it failing on linux too ?

**Comment by stefins** [comment]
Yeah It was failing on ubuntu also

**Comment by tertsdiepraam** [comment]
Seems like the `cfg(feature = chmod)` should apply to the test signature instead of just the single line. Otherwise it will fail on all platforms that don't have chmod enabled. Maybe that fixes the test?

**Comment by stefins** [comment]
@tertsdiepraam done

**Review by tertsdiepraam — APPROVED** [review]
Looks like it worked! I think it looks good now but I'll leave the final check to Sylvestre

**Comment by sylvestre** [comment]
Not sure what you mean but it works here:

`Warning: Congrats! The gnu test tests/rm/isatty is no longer failing!`

**Comment by stefins** [comment]
This test https://github.com/uutils/coreutils/runs/8071421179?check_suite_focus=true

**Comment by sylvestre** [comment]
oh that, don't bother ;)
we have intermittent issues

**Comment by sylvestre** [comment]
well done ;)

**Pull request merged**